File: AppGenerateFiles/dbo/FrontModel/Niveles.js
//@ts-check
import { EntityClass } from "../../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools, BasicStates } from "../../../WDevCore/WModules/WComponentsTools.js";
//@ts-ignore
import { ModelProperty } from "../../../WDevCore/WModules/CommonModel.js";
import { Asignaturas }  from './Asignaturas.js'
import { Clases }  from './Clases.js'
class Niveles extends EntityClass {
   /** @param {Partial<Niveles>} [props] */
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id;
   /**@type {String}*/ Nombre;
   /**@type {String}*/ Nombre_corto;
   /**@type {String}*/ Nombre_grado;
   /**@type {Number}*/ Numero_grados;
   /**@type {String}*/ Observaciones;
   /**@type {Boolean}*/ Habilitado;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Number}*/ Orden;
   /**@type {Number}*/ Inicio_grado;
   /**@type {Array<Asignaturas>} OneToMany*/ Asignaturas;
   /**@type {Array<Clases>} OneToMany*/ Clases;
}
export { Niveles }

File: AppGenerateFiles/dbo/Model/Niveles.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Niveles : EntityClass {
       [PrimaryKey(Identity = false)]
       public int? Id { get; set; }
       public string? Nombre { get; set; }
       public string? Nombre_corto { get; set; }
       public string? Nombre_grado { get; set; }
       public int? Numero_grados { get; set; }
       public string? Observaciones { get; set; }
       public bool? Habilitado { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       public int? Orden { get; set; }
       public int? Inicio_grado { get; set; }
       [OneToMany(TableName = "Asignaturas", KeyColumn = "Id", ForeignKeyColumn = "Nivel_id")]
       public List<Asignaturas>? Asignaturas { get; set; }
   }
}
